### PR TITLE
Remove course syllabus block (without compiled assets)

### PR DESCRIPTION
**Pull request by seothemes**
## Description

This PR removes the Course Syllabus block since an updated version has been added to the core LifterLMS plugin.

## How has this been tested?

- Manually tested
- PHPUnit

## Types of changes

Breaking change - This could cause issues if a site has an old version of the core LifterLMS plugin without the new syllabus block.

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests.
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

**Comment by seothemes** [comment]
Related https://github.com/gocodebox/lifterlms/pull/2374

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]
Thanks for this @seothemes 
Looking at it, though, since it's an extending class, I think we need to override all the public and protected methods and deprecated them as well, what do you think?

**Comment by seothemes** [comment]
@eri-trabiccolo I have made these changes if you'd like to take a look https://github.com/gocodebox/lifterlms-blocks/pull/187/commits/622de3f9ce8edea2b8fa150d4476d040090b70aa

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]
I think it's good.
Do you have the time to fix the failing PHP unit tests?
I think you can just get rid of the tests on this "component".

**Review by eri-trabiccolo — APPROVED** [review]

File: includes/blocks/class-llms-blocks-course-syllabus-block.php [hunk 1]
```diff
@@ -27,34 +26,40 @@ class LLMS_Blocks_Course_Syllabus_Block extends LLMS_Blocks_Abstract_Block {
...
-	 * @param   array  $attributes Optional. Block attributes. Default empty array.
-	 * @param   string $content    Optional. Block content. Default empty string.
-	 * @return  void
-	 * @since   1.0.0
-	 * @version 1.1.0
+	 * @since 1.0.0
+	 * @deprecated [version]
+	 *
+	 * @param array  $attributes Optional. Block attributes. Default empty array.
+	 * @param string $content Optional. Block content. Default empty string.
```
Comment (eri-trabiccolo) [inline includes/blocks/class-llms-blocks-course-syllabus-block.php]:
```suggestion
	 * @param array  $attributes Optional. Block attributes. Default empty array.
	 * @param string $content    Optional. Block content. Default empty string.
```

File: includes/blocks/class-llms-blocks-course-syllabus-block.php [hunk 2]
```diff
@@ -66,6 +71,105 @@ public function get_attributes() {
...
+	public function get_render_hook() {
+		llms_deprecated_function( __METHOD__, '[version]' );
 
-return new LLMS_Blocks_Course_Syllabus_Block();
+		return sprintf( '%1$s_%2$s_block_render', $this->vendor, $this->id );
+	}
+
+	/**
+	 * Removed hooks stub.
+	 * Extending classes can use this class to remove hooks attached to the render function action.
```
Comment (eri-trabiccolo) [inline includes/blocks/class-llms-blocks-course-syllabus-block.php]:
```suggestion
	 * Removed hooks stub.
	 *
	 * Extending classes can use this class to remove hooks attached to the render function action.
```

File: includes/blocks/class-llms-blocks-course-syllabus-block.php [hunk 3]
```diff
@@ -66,6 +71,105 @@ public function get_attributes() {
...
+	}
+
+	/**
+	 * Renders the block type output for given attributes.
+	 *
+	 * @since 1.0.0
+	 * @deprecated [version]
+	 *
+	 * @param array  $attributes Optional. Block attributes. Default empty array.
+	 * @param string $content Optional. Block content. Default empty string.
```
Comment (eri-trabiccolo) [inline includes/blocks/class-llms-blocks-course-syllabus-block.php]:
```suggestion
	 * @param array  $attributes Optional. Block attributes. Default empty array.
	 * @param string $content    Optional. Block content. Default empty string.
```

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]

**Review by eri-trabiccolo — APPROVED** [review]

**Pull request merged**